WPFHospitalEditor/EquipmentRelocation.xaml.cs

```diff
@@ -0,0 +1,179 @@
+            GetRoomsWithGivenEquipmentAmount();
+            SetRoomsComboBox();
+            this.DataContext = this;
+        }
+
+        private void CloseClick(object sender, RoutedEventArgs e)
+        {
+            this.Close();
+        }
+
+        private void SetRoomsComboBox()
+        {
+            List<Room> rooms = roomServerController.getRoomsByEquipmentType(relocationEquipmentName).ToList();
+            foreach (Room room in rooms)
+            {
+                if(room.Id != destinationRoomId)
+                    roomSearchComboBox.Items.Add(room.Id);
+            }
+        }
+
+        private void RelocateEquipment(object sender, RoutedEventArgs e)
+        {
+            startDate =
+                        DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy")
+                        + " " + StartTime.Text, "MM/dd/yyyy HH:mm", null);
+            endDate =
+                DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy")
+                + " " + EndTime.Text, "MM/dd/yyyy HH:mm", null);
+
+            if (CheckAmount() && !CheckIfDatesAreEqual(new TimeInterval(startDate, endDate)))
+            {
+                List<int> unavailableRooms = checkRoomsAvailability();
+                if (unavailableRooms.Count > 0)
+                {
+                    ShowAlternativeRelocationAppointments(unavailableRooms);
+                }
+                else
+                {
+                    ScheduleRelocation();
+                }
+            }
+
+        }
+
+        private void ShowAlternativeRelocationAppointments(List<int> unavailableRooms)
+        {
+            AlternativeRelocationAppointments newWindow =
+                        new AlternativeRelocationAppointments(unavailableRooms[0], hospitalMap, this);
+            newWindow.Show();
+        }
+
+        private void ScheduleRelocation()
+        {
+            while (startDate < endDate)
+            {
+                examinationServerController.ScheduleExamination(startDate, 1, 2);
+                startDate = startDate.AddMinutes(30);
+            }
+            MessageBox.Show("Relocation is successfully scheduled!", "");
+            this.Close();
+        }
+
+        private bool CheckIfDatesAreEqual(TimeInterval timeInterval)
+        {
+            if (timeInterval.Start >= timeInterval.End) 
+            {
+                MessageBox.Show("End time must be after start time!", "");
+                return true;
+            }
+            return false;
+        }
+
+        private List<int> checkRoomsAvailability()
+        {
+            DateTime startDate = DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy") + " " + StartTime.Text, "MM/dd/yyyy HH:mm", null);
+            DateTime endDate = DateTime.ParseExact(startDatePicker.SelectedDate.Value.ToString("MM/dd/yyyy") + " " + EndTime.Text, "MM/dd/yyyy HH:mm", null);
```

Comment:
Don't need to parse again. You parsed dates before calling this method. Just add dates as parameters.

Reply:
Fixed.